src/java.base/share/classes/jdk/internal/vm/vector/VectorSupport.java

```diff
@@ -621,6 +625,26 @@ VOUT convert(int oprId,
 
     /* ============================================================================ */
 
+    public interface ComExpOperation<V extends Vector<?>,
+                                     M extends VectorMask<?>> {
+        V apply(V v1, V v2, M m);
+    }
+
+    @IntrinsicCandidate
+    public static
+    <V extends Vector<E>,
+     M extends VectorMask<E>,
+     E>
+    V comExpOp(int opr,
```

Comment:
I prefer boolean instead of int.

Reply:
I have kept the implementation consistent with rest of vector API code and passing an Opcode instead of flag to differentiate between compress/expand operation